### PR TITLE
chore: revert export provider context

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/@react-spectrum/color/test/ColorWheel.test.tsx [hunk 1]
```diff
@@ -341,7 +341,7 @@ describe('ColorWheel', () => {
       let defaultColor = parseColor('hsl(0, 100%, 50%)');
       let {container: _container, getByRole} = render(<ControlledHSL defaultValue={defaultColor} onChange={onChangeSpy} onChangeEnd={onChangeEndSpy} />);
       let slider = getByRole('slider');
-      let container = _container?.firstChild?.firstChild?.firstChild as HTMLElement;
+      let container = _container?.firstChild?.firstChild?.firstChild?.firstChild as HTMLElement;
```
Comment (devongovett) [inline packages/@react-spectrum/color/test/ColorWheel.test.tsx:344]: 😂

Comment (snowystinger) [reply packages/@react-spectrum/color/test/ColorWheel.test.tsx:344]: yeah... we have far too many of these, they are very silly

File: packages/dev/test-utils/src/renderOverride.js [hunk 2]
```diff
@@ -34,3 +37,7 @@ function customRender(ui, options) {
 
 // override render method with
 export {customRender as render};
+
+export function renderv3(ui, options, providerProps) {
```
Comment (devongovett) [inline packages/dev/test-utils/src/renderOverride.js:41]: just confirming: this isn't the public test utils package right?

Comment (snowystinger) [reply packages/dev/test-utils/src/renderOverride.js:41]: correct, this is private

File: packages/@react-aria/focus/src/FocusScope.tsx [hunk 3]
```diff
@@ -692,7 +692,7 @@ function useRestoreFocus(scopeRef: RefObject<Element[] | null>, restoreFocus?: b
         restoreFocus
         && nodeToRestore
         && (
-          // eslint-disable-next-line react-hooks/exhaustive-deps
+
```
Comment (devongovett) [inline packages/@react-aria/focus/src/FocusScope.tsx:695]:
```suggestion
```

**Review by devongovett — DISMISSED** [review]

File: packages/dev/test-utils/src/renderOverride.js (hunk 2 above)
Comment (snowystinger) [reply packages/dev/test-utils/src/renderOverride.js:41]: https://github.com/adobe/react-spectrum/blob/main/packages/dev/test-utils/package.json#L4

File: packages/@react-aria/focus/src/FocusScope.tsx (hunk 3 above)
Comment (snowystinger) [reply packages/@react-aria/focus/src/FocusScope.tsx]: silly auto fix

**Review by devongovett — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**